modules/multiqc/main.nf

```diff
@@ -1,17 +1,21 @@
 params.outdir = 'results'
 
-process MULTIQC {
-    conda 'multiqc=1.17'
-    publishDir params.outdir, mode:'copy'
-
-    input:
-    path('*') 
-    path(config) 
-
-    output:
-    path('multiqc_report.html')
-
-    script:
+@ProcessFn(
+    directives={
+        conda 'multiqc=1.17'
+        publishDir params.outdir, mode:'copy'
+    },
+    inputs={
+        path { files }, stageAs: '*'
+        path { config }
+    },
+    outputs={
+        path '$file0', 'multiqc_report.html'
+        emit { path('$file0') }
+    },
+    script=true
+)
+def MULTIQC(List<Path> files, Path config) {
```

Comment:
I appreciate the typing on input channels. It makes me wonder if it would be possible to add a decorator like this:

```
    input:
    @type:List<Path>
    @description: a list of files output from various processes for multiqc to parse
    path('*') 
    
    output:
    @type:Path
    @description: multiqc html summary
    path('multiqc_report.html')
```
To the current syntax that could be both checked, and generate some automatic documentation. It would be especially nice if the type of channel -- value or queue -- was included. This would be helpful in development and maintenance, I think.

Reply:
The syntax would have to be something like this:
```groovy
    input:
    path('*'), type: List<Path>, description: 'a list of files output from various processes for multiqc to parse'
```

But you have the right idea.

As for channels, I think the process definition should be decoupled from whether the input channels are queue or value, because it shouldn't really matter